lib/ip.js

```diff
@@ -232,7 +233,18 @@ binet.write = function write(dst, str, off, size) {
 
 binet.writeBW = function writeBW(bw, str, size) {
   assert(bw && typeof bw === 'object');
-  bw.offset = binet.write(bw.data, str, bw.offset, size);
+
+  // StaticWriter
+  if (bw.data) {
```

Comment:
Is this the best way to identify StaticWriter? Just wondering if `instanceof` or whatever would be more clear. Also do you think the jsdoc `@param` should be expanded as well?

Reply:
That was how it was identifying StaticWriter, just there was no documentation about it and it had generalized "BufferWriter" form there. In general APIs are same, so they can be used w/o noticing what you are using. But this one was using this hack for more performance, but missed normal BufferWriter. Normal BufferWriter is not used anywhere for now.

No, in that case we will have to introduce dependency, and even then if there's version difference in the dependencies we will still end up having instanceof not matching. Unfortunately, this is best we can do with js...